pkg/collector/corechecks/containers/containerd.go

```diff
@@ -123,10 +124,10 @@ func (c *ContainerdCheck) Run() error {
 		// Process events
 		computeEvents(events, sender, c.filters)
 	}
-
-	nk := namespaces.WithNamespace(context.Background(), ns)
+	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
```

Comment:
probably worth it to make it configurable like https://github.com/DataDog/datadog-agent/blob/33db6c888082da73e04a7d344b9c78ee3a72371d/pkg/util/containers/cri/util.go#L80

Reply:
Some of the calls already had the timeout from this option (notably `Containers()`).
I think it makes sense to be consistent across all calls and therefore defer the handling of the timeout to the util.

To this extent I made this https://github.com/DataDog/datadog-agent/pull/3101/commits/aa30bf653b90bbc0d53340d0cb06727f74e811be

The cleanest option I can think of without refactoring this logic is to add this timeout value to the ContainerdCheck struct and use it later on. It avoids exporting the one from the ContainerdUtil but it duplicates some code.

Let me know your thoughts